packages/c/chipmunk2d/xmake.lua

```diff
@@ -38,7 +49,13 @@ package("chipmunk2d")
             table.insert(configs, "-DBUILD_STATIC=ON")
             table.insert(configs, "-DINSTALL_STATIC=ON")
         end
-        import("package.tools.cmake").install(package, configs)
+        local opt = {}
+        if package:config("precision") == "double" then
+            opt.cxflags = compiler.map_flags("c", "define", "CP_USE_DOUBLES=1", {target = self})
```

Comment:
I think so, but why hardcode a compiler flag instead of using flag mpaping which is more resilient if we were to use a C/C++ toolchain not using GCC-like flags?

Reply:
Because I didn't want to overcomplicate the install script, it's a bit complicated to go through the extra step of importing the compiler module and calling compiler.map_flags just to set up a `macro`. This can lead to package scripts that are hard to maintain. Also, I don't recommend using the compiler.map_flags interface on the user side.

Reply:
Done